### DISCUSSION TITLE
this acme.sh is not changing in anyway file permissions?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [acmesh-official/acme.sh#5122] [General]
I was wondering if this acme.sh is not changing in anyway file permission. As in if I set specific rw on group or with setfacl? Because it is now 2nd time I need to correct file permissions on files, otherwise services don't start.

**Comment by neilpang** [comment]
acme.sh only tightens permissions on its own internal storage (`~/.acme.sh` -- keys/configs are chmod 600 at creation, by design). Do not point services there.

The `--install-cert` target files are updated in place with `cat >`, so existing permissions/ACLs on them are preserved across renewals; `chmod 600` is applied to the key target only when acme.sh creates that file for the first time. Install the cert to service-owned paths once, set your group/ACL there, and renewals will not touch it.
